### PR TITLE
update: bulk resolve remaining issues in plugin framework

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind bug

/kind cleanup

**Any specific area of the project related to this PR?**

/area libsinsp

/area tests

**Does this PR require a change in the driver versions?**

Don't think so.

**What this PR does / why we need it**:

This PR contributes a series of fixes and improvements to the plugin framework, following up the recent changes and feature additions:
- Remove whitespace from `plugin ID` param of the `PPME_PLUGINEVENT_E` event
- Improve error reporting on plugin loader, with specific attention on when capabilities are implemented incompletely or with forbidden values
- Plugin manager has been updated to be resilient on plugin with event sourcing capability but no ID (newly introduced notion for generating any event from the libscap event table)
- Fixed documentation affected miscs code parts (e.g. `evt.pluginnname` and `evt.plugininfo` fields)
- Support error reporting for cross-plugin state table access
- Removed failed attempt of passing numeric ID of event's source through plugin API (linked issue: https://github.com/falcosecurity/libs/issues/1097)
- Tests, tests, tests

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: bulk resolve remaining issues in plugin framework
fix(driver)!: fix plugin ID field name in PPME_PLUGINEVENT_E
```

### CODE REVIEW

File: driver/event_table.c [hunk 1]
```diff
@@ -371,7 +371,7 @@ const struct ppm_event_info g_event_info[] = {
 	[PPME_SYSCALL_RENAMEAT2_X] = {"renameat2", EC_FILE | EC_SYSCALL, EF_NONE, 6, {{"res", PT_ERRNO, PF_DEC}, {"olddirfd", PT_FD, PF_DEC}, {"oldpath", PT_FSRELPATH, PF_NA, DIRFD_PARAM(1)}, {"newdirfd", PT_FD, PF_DEC}, {"newpath", PT_FSRELPATH, PF_NA, DIRFD_PARAM(3)}, {"flags", PT_FLAGS32, PF_HEX, renameat2_flags} } },
 	[PPME_SYSCALL_USERFAULTFD_E] = {"userfaultfd", EC_FILE | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 0},
 	[PPME_SYSCALL_USERFAULTFD_X] = {"userfaultfd", EC_FILE | EC_SYSCALL, EF_CREATES_FD | EF_MODIFIES_STATE, 2, {{"res", PT_ERRNO, PF_DEC}, {"flags", PT_FLAGS32, PF_HEX, file_flags} } },
-	[PPME_PLUGINEVENT_E] = {"pluginevent", EC_OTHER | EC_PLUGIN, EF_LARGE_PAYLOAD, 2, {{"plugin ID", PT_UINT32, PF_DEC}, {"event_data", PT_BYTEBUF, PF_NA} } },
+	[PPME_PLUGINEVENT_E] = {"pluginevent", EC_OTHER | EC_PLUGIN, EF_LARGE_PAYLOAD, 2, {{"plugin_id", PT_UINT32, PF_DEC}, {"event_data", PT_BYTEBUF, PF_NA} } },
```
Comment (FedeDP) [inline driver/event_table.c:374]: Nit: this is a somewhat breaking change; make sure to `!` highlight it :D

**Review by FedeDP — APPROVED** [review]
Nice work Jason! This LGTM!
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**